src/stacks-hierarchy/Root_TabNavigatorStack/TabNav_DashboardStack/Dashboard_TripSearchScreen/components/CityZoneMessage.tsx

```diff
@@ -40,39 +44,66 @@ export const CityZoneMessage: React.FC<CityZoneMessageProps> = ({
   const fromCityZone = useFindCityZoneInLocation(from, cityZones);
   const toCityZone = useFindCityZoneInLocation(to, cityZones);
 
-  if (!fromCityZone || !toCityZone) {
+  if (!fromCityZone?.enabled || !toCityZone?.enabled) {
     return null;
   }
 
   if (fromCityZone.id !== toCityZone.id) {
     return null;
   }
 
-  const openUrlForCityZone = (cityZone: CityZone) => {
-    const contactUrl = getTextForLanguage(cityZone.contactUrl, language);
-    if (contactUrl) {
-      Linking.openURL(contactUrl);
-    }
-  };
+  const actionButtons: ActionButton[] = [];
 
-  const messageActions = [
-    {
-      id: `${fromCityZone.id}_action`,
-      text: fromCityZone.name,
-      onPress: () => openUrlForCityZone(fromCityZone),
-    },
-  ];
+  const orderUrl = getTextForLanguage(fromCityZone.orderUrl, language);
+  if (orderUrl) {
+    actionButtons.push({
+      id: `book_online_action`,
+      text: t(CityBoxMessageTexts.actionButtons.bookOnline),
+      icon: ExternalLink,
+      interactiveColor: 'interactive_0',
+      accessibilityHint: t(CityBoxMessageTexts.a11yHintForExternalContent),
+      onPress: () => Linking.openURL(orderUrl),
+    });
+  }
 
-  return (
-    <Section style={style.cityZoneMessage}>
-      <CityZoneBox
-        message={t(CityBoxMessageTexts.message)}
-        icon={() => <FlexibleTransport />}
-        onDismiss={onDismiss}
-        actionButtons={messageActions}
-      />
-    </Section>
-  );
+  const phoneNumber = fromCityZone.phoneNumber;
+  if (phoneNumber && !orderUrl) {
+    actionButtons.push({
+      id: `book_by_phone_action`,
+      icon: Phone,
+      text: t(CityBoxMessageTexts.actionButtons.bookByPhone),
+      interactiveColor: 'interactive_0',
+      accessibilityHint: t(CityBoxMessageTexts.a11yHintForPhone),
+      onPress: () => Linking.openURL(`tel:${phoneNumber}`),
+    });
+  }
+
+  const moreInfoUrl = getTextForLanguage(fromCityZone.moreInfoUrl, language);
+  if (moreInfoUrl && moreInfoUrl) {
+    actionButtons.push({
+      id: `more_info_action`,
+      icon: ExternalLink,
+      text: t(CityBoxMessageTexts.actionButtons.moreInfo),
+      interactiveColor: 'interactive_3',
+      accessibilityHint: t(CityBoxMessageTexts.a11yHintForExternalContent),
+      onPress: () => Linking.openURL(moreInfoUrl),
+    });
+  }
+
+  if (actionButtons.length > 0) {
+    return (
+      <Section style={style.cityZoneMessage}>
+        <CityZoneBox
+          message={t(CityBoxMessageTexts.message(fromCityZone.name))}
+          icon={() => <FlexibleTransport />}
+          onDismiss={onDismiss}
+          actionButtons={actionButtons}
```

Comment:
I found it a bit hard to understand the relation between `actionButtons` and all the if checks above. 
Could it be an option to create a function `getActionButtons` that contains all the above if's `(if (phoneNumber && !orderUrl)`, `if (orderUrl)` ... and returns the list of buttons?

Reply:
I made fixes based on your suggestions